designs/Prerelease-Option.md

```diff
@@ -0,0 +1,55 @@
+# Spec Name
+
+* Status: In Review
+* Author(s): [Martin Ruiz](https://github.com/martinrrm)
+* Issue: [4699](https://github.com/NuGet/Home/issues/4699) dotnet add package - support for NuGet "Pre" version
+
+## Problem Background
+
+Customers want to add the latest version of a package to their project, even tough it is a prerelease version.
+
+## Who are the customers
+
+All dotnet.exe customers
+
+## Goals
+
+Add an option to the command `add package` enable installing latest prerelease version.
+
+## Non-Goals
+
+## Solution
+
+Add the option `--prerelease` in the command `add package`. This command should get the latest version of the packet available even if it is not a stable version.
+
+```package 2.0.0
+package 3.0.0
+package 3.3.1-preview.3
+```
+
+| Command | Result | Description |
+|---------|--------|--------------|
+| `dotnet.exe add package` | 3.0.0 | latest stable version of package |
+| `dotnet.exe add package --prerelease` | 3.3.1-preview.3 | latest version of package |
+| `dotnet.exe add package --prerelease --version 3.0.0` | error | The user cannot use this commands at the same time |
+
+```package 2.0.0
```

Comment:
nit: 
when you format put it on a new line.

Reply:
A couple of questions/observations:
1. Using the —prerelease option should get the latest version including prerelease. Correct?
2. I find the name of the option a bit long. Can we also enable —pre may in addition to —prerelease? - a nit

Reply:
1. Yes. Similar to how the PM UI behaves. If you have the prerelease filter we show you the latest version if it's prerelease. 

2. We just opted for the full name cause I haven't seen option names not being the exact word. We can't really do --pre as  short hand. A short option must be one letter only. 
So we either have `--prerelease` or `--pre`.

Reply:
> 1. Yes. Similar to how the PM UI behaves. If you have the prerelease filter we show you the latest version if it's prerelease.
> 2. We just opted for the full name cause I haven't seen option names not being the exact word. We can't really do --pre as  short hand. A short option must be one letter only.
>    So we either have `--prerelease` or `--pre`.

Personally, I'd rather go with the options that is the most instinctual. So if I forgot what the options were but I wanted to enable prereleases, my instinct would be to add `--prerelease` not `--pre` even tho it's shorter. I think intuitiveness > saving a couple key strokes.

Reply:
I vote sticking with `--prerelease` because it's the most instinctive and is the wording used in other client tools such as PMUI and nuget.exe. If I forget what the options are and want to install a prerelease package, my instinct would be to try `--prerelease` before `--pre`.